### PR TITLE
filtering by trainer

### PR DESCRIPTION
**Pull request by cvbelk**

### CODE REVIEW

File: ormconfig.js [hunk 1]
```diff
@@ -2,5 +2,7 @@ module.exports = {
     "type": "postgres",
     "url": process.env.DB_URL,
     "entities": [process.env.ENTITY_PATH],
-    "synchronize": false
+    "synchronize": false,
+    //"logging": true,
```
Comment (Bortnyak) [inline ormconfig.js]: Why we no need the logging?

File: src/DAO/scheduleDAO.ts [hunk 2]
```diff
@@ -21,6 +33,45 @@ export class ScheduleDAO
...
+    const train = options.trainer;
+    const time = options.byTime;
+    const srtBy = options.sortBy;
+    const srt = options.sort;
+    const fromDate = new Date(frDate);
+    const untilDate = new Date(tilDate);
+    const searchParams = {
+      trainer: train,
+      startDate: Between(fromDate, untilDate),
+      endDate: Raw(
```
Comment (Bortnyak) [inline src/DAO/scheduleDAO.ts]: as we've discussed this is unnecessary

File: src/DAO/scheduleDAO.ts [hunk 3]
```diff
@@ -21,6 +33,45 @@ export class ScheduleDAO
...
+  async findByFilters(options: ISearchParams): Promise<Schedule[]> {
+    const scheduleRepository = await this._getRepository(ScheduleEntity);
+    const frDate = options.fromDate;
+    const tilDate = options.untilDate;
+    const train = options.trainer;
+    const time = options.byTime;
+    const srtBy = options.sortBy;
+    const srt = options.sort;
+    const fromDate = new Date(frDate);
+    const untilDate = new Date(tilDate);
```
Comment (Bortnyak) [inline src/DAO/scheduleDAO.ts]: frDate, tilDate, train, time, srtBy, srt, fromDate, untilDate - redundant variables. Can we simplify that? What about one more method in the service layer, to pick up only existing filters?

File: src/DAO/scheduleDAO.ts [hunk 4]
```diff
@@ -21,6 +33,45 @@ export class ScheduleDAO
...
+      trainer: train,
+      startDate: Between(fromDate, untilDate),
+      endDate: Raw(
+        (alias) =>
+          `to_char(timestamp '${time}', 'HH24:MI') BETWEEN to_char(ScheduleEntity.startDate, 'HH24:MI') AND to_char(${alias}, 'HH24:MI')`,
+      ),
+    };
+    const orderParams: OrderByCondition = {};
+    if (srt && srtBy) {
+      orderParams[srtBy] = srt === 'ASC' ? 'ASC' : 'DESC';
```
Comment (Bortnyak) [inline src/DAO/scheduleDAO.ts]: see above comments

File: src/DAO/scheduleDAO.ts [hunk 5]
```diff
@@ -21,6 +33,45 @@ export class ScheduleDAO
...
+    }
+    if (!train) {
+      delete searchParams.trainer;
+    }
+    if (!frDate || !tilDate) {
+      delete searchParams.startDate;
+    }
+    if (!time) {
+      delete searchParams.endDate;
```
Comment (Bortnyak) [inline src/DAO/scheduleDAO.ts]: all these if's are redundant

**Review by Bortnyak — CHANGES_REQUESTED** [review]

File: ormconfig.js (hunk 1 above)
Comment (cvbelk) [reply ormconfig.js]: deleted

File: src/DAO/scheduleDAO.ts (hunk 2 above)
Comment (cvbelk) [reply src/DAO/scheduleDAO.ts]: byTime filter deleted

File: src/DAO/scheduleDAO.ts (hunk 3 above)
Comment (cvbelk) [reply src/DAO/scheduleDAO.ts]: maybe not in service layer(That injects DAO), how about private helper-method in here

File: src/DAO/scheduleDAO.ts [hunk 6]
```diff
@@ -21,6 +35,42 @@ export class ScheduleDAO
     return await scheduleRepository.find({ relations: ['trainer', 'class'] });
   }
 
+  private prepareSort(options: ISearchParams): OrderByCondition {
+    const orderParams: OrderByCondition = {};
+    if (options.sort && options.sortBy) {
+      orderParams[options.sortBy] = options.sort === 'ASC' ? 'ASC' : 'DESC';
+    }
```
Comment (Bortnyak) [inline src/DAO/scheduleDAO.ts:42]:
```js
if (options.sortBy) {
   orderParams[options.sortBy] = options.sort ?? 'ASC';
}
```
also would be nice to move the default sorting method to constant

File: src/DAO/scheduleDAO.ts [hunk 7]
```diff
@@ -21,6 +35,42 @@ export class ScheduleDAO
...
+    return orderParams;
+  }
+  private prepareFilters(
+    options: ISearchParams,
+  ): { trainer: User; startDate: FindOperator<Date> } {
+    const searchParams = {
+      trainer: options.trainer,
+      startDate: Between(options.fromDate, options.untilDate),
+    };
+    if (!options.trainer) {
```
Comment (Bortnyak) [inline src/DAO/scheduleDAO.ts:52]: these "if's" could be simplified:


```js
const searchParams: ISearchParams = {};

Object.keys(options).map(key => {  
  if (searchParams[key]) {
    obj[key] = filtersToApply[key];
  } 
});

```

something along those lines (not sure it's working code)

**Review by Bortnyak — CHANGES_REQUESTED** [review]

File: src/DAO/scheduleDAO.ts (hunk 6 above)
Comment (cvbelk) [reply src/DAO/scheduleDAO.ts:42]: Done

**Comment by Bortnyak** [comment]
closed

**Pull request closed without merging**